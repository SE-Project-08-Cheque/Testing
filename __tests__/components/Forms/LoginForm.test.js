// import dependencies
import React from "react";

// import react-testing methods
import { render, fireEvent, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import LoginForm from "../../../components/Forms/LoginForm";

afterEach(cleanup);


it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <LoginForm />,
    div
  );
});

it("Log in correctly", () => {
  const { getByTestId } = render(
    <LoginForm />
  );

 
  fireEvent.change(getByTestId("email"),{ target: { value: 'chandima@gmail.com' } })
  fireEvent.change(getByTestId("password"),{ target: { value: 'amarasena' } })
 
  
  fireEvent.click(getByTestId("login")) 

  
});

it("matches snapshot", () => {
  const tree = renderer.create(
    <LoginForm/>
  );

  expect(tree).toMatchSnapshot();
});
