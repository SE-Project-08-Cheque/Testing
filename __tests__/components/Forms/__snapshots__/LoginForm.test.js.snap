// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-mccp0p"
>
  <h4
    className="chakra-heading css-biiv2x"
  >
    Log In
  </h4>
  <form
    action="submit"
  >
    <div
      className="chakra-stack css-1n2jfo2"
    >
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <div
          className="chakra-input__group css-4302v8"
        >
          <div
            className="chakra-input__left-element css-j74x9m"
          >
            <svg
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <g
                fill="currentColor"
              >
                <path
                  d="M11.114,14.556a1.252,1.252,0,0,0,1.768,0L22.568,4.87a.5.5,0,0,0-.281-.849A1.966,1.966,0,0,0,22,4H2a1.966,1.966,0,0,0-.289.021.5.5,0,0,0-.281.849Z"
                />
                <path
                  d="M23.888,5.832a.182.182,0,0,0-.2.039l-6.2,6.2a.251.251,0,0,0,0,.354l5.043,5.043a.75.75,0,1,1-1.06,1.061l-5.043-5.043a.25.25,0,0,0-.354,0l-2.129,2.129a2.75,2.75,0,0,1-3.888,0L7.926,13.488a.251.251,0,0,0-.354,0L2.529,18.531a.75.75,0,0,1-1.06-1.061l5.043-5.043a.251.251,0,0,0,0-.354l-6.2-6.2a.18.18,0,0,0-.2-.039A.182.182,0,0,0,0,6V18a2,2,0,0,0,2,2H22a2,2,0,0,0,2-2V6A.181.181,0,0,0,23.888,5.832Z"
                />
              </g>
            </svg>
          </div>
          <input
            aria-describedby="field-13-helptext"
            aria-label="Email"
            aria-required={true}
            className="chakra-input css-0"
            data-testid="email"
            disabled={false}
            id="field-13"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Email"
            readOnly={false}
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="chakra-form__helper-text css-1xnfdvb"
          id="field-13-helptext"
        >
          
        </div>
      </div>
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <div
          className="chakra-input__group css-4302v8"
        >
          <div
            className="chakra-input__left-element css-j74x9m"
          >
            <svg
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M19.5,9.5h-.75V6.75a6.75,6.75,0,0,0-13.5,0V9.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5ZM7.75,6.75a4.25,4.25,0,0,1,8.5,0V9a.5.5,0,0,1-.5.5H8.25a.5.5,0,0,1-.5-.5Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <input
            aria-describedby="field-14-helptext"
            aria-label="Password"
            aria-required={true}
            className="chakra-input css-0"
            data-testid="password"
            disabled={false}
            id="field-14"
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Password"
            readOnly={false}
            required={true}
            type="password"
            value=""
          />
          <div
            className="chakra-input__right-element css-1feb43d"
          >
            <button
              className="chakra-button css-d5t03s"
              onClick={[Function]}
              type="button"
            >
              Show
            </button>
          </div>
        </div>
        <div
          className="chakra-form__helper-text css-1xnfdvb"
          id="field-14-helptext"
        >
          
        </div>
      </div>
      <button
        className="chakra-button css-oqfhsh"
        data-testid="login"
        onClick={[Function]}
        type="button"
      >
        Log In
      </button>
    </div>
  </form>
</div>
`;
